internal/api/verify.go

```diff
@@ -392,13 +404,25 @@ func (a *API) prepErrorRedirectURL(err *HTTPError, w http.ResponseWriter, r *htt
 	}
 	q.Set("error_code", strconv.Itoa(err.Code))
 	q.Set("error_description", err.Message)
-	return rurl + "#" + q.Encode()
+	if flowType == models.PKCEFlow {
+		u.RawQuery = q.Encode()
+	}
+	u.Fragment = q.Encode()
+	return u.String(), nil
 }
 
-func (a *API) prepRedirectURL(message string, rurl string) string {
-	q := url.Values{}
-	q.Set("message", message)
-	return rurl + "#" + q.Encode()
+func (a *API) prepRedirectURL(message string, rurl string, flowType models.FlowType) (string, error) {
+	u, perr := url.Parse(rurl)
+	if perr != nil {
+		return "", perr
+	}
+	q := u.Query()
+	if flowType == models.PKCEFlow {
+		q.Set("message", message)
+	}
+	u.RawQuery = q.Encode()
+	u.Fragment = fmt.Sprintf("message=%s", url.QueryEscape(message))
```

Comment:
We don't  use `q.Encode()` as there might be existing query params in the url which we don't want to pass into the fragment